src/AppInstallerCLIE2ETests/ShowCommand.cs

```diff
@@ -50,29 +50,49 @@ public void ShowWithNoMatches()
+            var result = TestCommon.RunAICLICommand("show", $"AppInstallerTest -s {ShowTestSourceName}");
             Assert.AreEqual(Constants.ErrorCode.ERROR_NO_APPLICATIONS_FOUND, result.ExitCode);
             Assert.True(result.StdOut.Contains("No package found matching input criteria."));
         }
 
         [Test]
-        public void ShowWithSingleMatch()
+        public void ShowWithNameMatch()
         {
-            // Show with 1 search match shows detailed manifest info
-            var result = TestCommon.RunAICLICommand("show", $"Microsoft.VisualStudioCode -s {ShowTestSourceName}");
+            var result = TestCommon.RunAICLICommand("show", $"--name TestExampleInstaller -s {ShowTestSourceName}");
+            Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("Found TestExampleInstaller [AppInstallerTest.TestExampleInstaller]"));
+            Assert.True(result.StdOut.Contains("TestExampleInstaller"));
+            Assert.True(result.StdOut.Contains("AppInstallerTest.TestExampleInstaller"));
+        }
+
+        [Test]
+        public void ShowWithIDMatch()
+        { 
+            var result = TestCommon.RunAICLICommand("show", $"--id AppInstallerTest.TestExampleInstaller -s {ShowTestSourceName}");
             Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
-            Assert.True(result.StdOut.Contains("Microsoft.VisualStudioCode"));
-            Assert.True(result.StdOut.Contains("Visual Studio Code"));
+            Assert.True(result.StdOut.Contains("Found TestExampleInstaller [AppInstallerTest.TestExampleInstaller]"));
+            Assert.True(result.StdOut.Contains("TestExampleInstaller"));
+            Assert.True(result.StdOut.Contains("AppInstallerTest.TestExampleInstaller"));
         }
 
         [Test]
         public void ShowWithVersions()
         {
             // Show with --versions list the versions
-            var result = TestCommon.RunAICLICommand("show", $"Microsoft.VisualStudioCode --versions -s {ShowTestSourceName}");
+            var result = TestCommon.RunAICLICommand("show", $"TestExampleInstaller --versions -s {ShowTestSourceName}");
             Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
-            Assert.True(result.StdOut.Contains("Microsoft.VisualStudioCode"));
-            Assert.True(result.StdOut.Contains("1.41.1"));
+            Assert.True(result.StdOut.Contains("TestExampleInstaller"));
+            Assert.True(result.StdOut.Contains("AppInstallerTest.TestExampleInstaller"));
+            Assert.True(result.StdOut.Contains("1.2.3.4"));
+        }
+
+        [Test]
+        public void ShowWithExactArg()
+        {
+            // Show 'testexampleinstaller' with exact arg should return none due to case sensitivity
+            var result = TestCommon.RunAICLICommand("show", $"--exact testexampleinstaller -s {ShowTestSourceName}");

```

Comment:
>show [](start = 53, length = 4)

can we also add a 'show with exact arg' test that exists in the repo?

Reply:
Added ShowWithExactName & ShowWithExactID to test that the "show --exact" command can find and show a test installer that exists in the repo.